crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -530,19 +530,65 @@ where
                 self.wr.write_str_lit(num.span, "-")?;
             }
             self.wr.write_str_lit(num.span, "Infinity")?;
-        } else if num.value.is_sign_negative() && num.value == 0.0 {
```

Comment:
We can't have negative number here, because `-100` is `UnaryExpression + NumericLiteral`